lib/Timber.php

```diff
@@ -124,67 +124,164 @@ class_alias( 'Timber\Timber', 'Timber' );
 	================================ */
 
 	/**
-	 * Get a post by post ID or query (as a query string or an array of arguments).
+	 * Gets a Timber Post by post ID or WP_Post object.
+	 *
+	 * By default, Timber will use the `Timber\Post` class to create a new post object. To control
+	 * which class is used for your post object, use [Class Maps]().
 	 *
 	 * @api
-	 * @deprecated since 2.0.0 Use `new Timber\Post()` instead.
+	 * @example
+	 * ```php
+	 * // Using a post ID.
+	 * $post = Timber::get_post( 75 );
 	 *
-	 * @param mixed        $query     Optional. Post ID or query (as query string or an array of
-	 *                                arguments for WP_Query). If a query is provided, only the
-	 *                                first post of the result will be returned. Default false.
-	 * @param string|array $PostClass Optional. Class to use to wrap the returned post object.
-	 *                                Default 'Timber\Post'.
+	 * // Using a WP_Post object.
+	 * $post = Timber::get_post( $wp_post );
+	 *
+	 * // Use currently queried post. Same as using get_the_ID() as a parameter.
+	 * $post = Timber::get_post();
+	 * ```
 	 *
-	 * @return \Timber\Post|bool Timber\Post object if a post was found, false if no post was
-	 *                           found.
+	 * @todo Add links to Class Maps documentation in function summary.
+	 * @todo Remove warnings in Timber 3.0
+	 *
+	 * @param null|int|\WP_Post $post Optional. Post ID or WP_Post object to get a Timber Post
+	 *                                object. Default `null`.
+	 *
+	 * @return \Timber\Post|bool `Timber\Post` object or an instance of a child class of
+	 *                           `Timber\Post` if a post was found, `false` if no post was found.
 	 */
-	public static function get_post( $query = false, $PostClass = 'Timber\Post' ) {
-		return PostGetter::get_post($query, $PostClass);
+	public static function get_post( $post = null ) {
+		if ( is_string( $post ) ) {
```

Comment:
sometimes we get IDs as strings, e.g. from `$_GET`. Factories support this already; should we support it here? We could add an `!is_numeric()` check to be certain that we definitely don't have an ID. Only consideration there would be if the `post_title` were a purely string...which would be weird and edge-casey. I feel like that's an okay trade-off...?

Reply:
Good call! Resolved in e5e0a4543b85b2110f0d69552e6fe6b16537ad4e.